test/js/bun/udp/udp.test.ts

```diff
@@ -0,0 +1,488 @@
+                validateRecv(socket, data, port, address, binaryType, bytes);
+
+                server.close();
+                client.close();
+                done();
+              },
+            },
+          });
+          client = await udpSocket({
+            connect: {
+              port: server.port,
+              hostname: server.hostname,
+            },
+          });
+
+          // handle unreliable transmission in UDP
+          function sendRec() {
+            if (!client.closed) {
+              validateSend(client.send(data));
+              setTimeout(sendRec, 10);
+            }
+          }
+          sendRec();
+        },
+        { timeout: 1000000 },
+      );
+
+      test(
+        `sendMany ${label} (${binaryType || "undefined"})`,
+        async done => {
+          const client = await udpSocket({});
+          let count = 0;
+          const server = await udpSocket({
+            binaryType: binaryType,
+            socket: {
+              data(socket, data, port, address) {
+                validateRecv(socket, data, port, address, binaryType, bytes);
+
+                count += 1;
+                if (count === 100) {
+                  server.close();
+                  client.close();
+                  done();
+                }
+              },
+            },
+          });
+
+          const payload = Array(100).fill([data, server.port, "127.0.0.1"]).flat();
+
+          // handle unreliable transmission in UDP
+          function sendRec() {
+            if (!client.closed) {
+              validateSendMany(client.sendMany(payload), 100);
+              setTimeout(sendRec, 10);
+            }
+          }
+          sendRec();
+        },
+        { timeout: 1000000 },
+      );
+
+      test(`sendMany connected ${label} (${binaryType || "undefined"})`, async done => {
+        // const client = await udpSocket({});
+        let client;
+        let count = 0;
+        const server = await udpSocket({
+          binaryType: binaryType,
+          socket: {
+            data(socket, data, port, address) {
+              validateRecv(socket, data, port, address, binaryType, bytes);
+
+              count += 1;
+              if (count === 100) {
+                server.close();
+                client.close();
+                done();
+              }
+            },
+          },
+        });
+
+        client = await udpSocket({
+          connect: {
+            port: server.port,
+            hostname: server.hostname,
+          },
+        });
+
+        const payload = Array(100).fill(data);
+
+        // handle unreliable transmission in UDP
+        function sendRec() {
+          if (!client.closed) {
+            validateSendMany(client.sendMany(payload), 100);
+            setTimeout(sendRec, 10);
+          }
+        }
+        sendRec();
+      });
+    }
+  }
+});
+
+describe("createSocket()", () => {
+  test("connect", done => {
+    const PORT = 12345;
```

Comment:
this will cause flaky tests, could we use `0`?

Reply:
the problem is that this test is supposed to test for the fact that binding a port work. not sure how to get around this without getting rid of the test